node/service/src/relay_chain_selection.rs

```diff
@@ -54,7 +54,7 @@ use std::sync::Arc;
 ///
 /// This is a safety net that should be removed at some point in the future.
 // Until it's not, make sure to also update `MAX_HEADS_LOOK_BACK` in `approval-voting`
-// when changing its value.
+// and `MAX_BATCH_SCRAPE_ANCESTORS` in `dispute-coordinator` when changing its value.
```

Comment:
nit: We should probably add these shared constants to a separate helper crate, and import it, to avoid these manual _keep in sync_-todos

Reply:
Agreed, do you have any suggestion on what would be the best place to move this ?

AFAIK we also want to remove this in the near future, so if we were to create a new separate crate just to hold these constants, it won't be a good investment.

Reply:
Removed this, we now scrape up to and including last finalized block.

Reply:
I think it's out of s ope for this PR, but a tracking issue would be nice

Reply:
The reason we have though lookup limits is to avoid a very busy (and wasteful) loop on startup if the node is syncing and is very much behind the tip of the chain. Imagine if we get the first leaf, but not finality notification. The difference between the could be millions of blocks in theory. However, I'm not sure how much of a problem it's in practice.

Reply:
I meant the const ecrraction, not the isdue you described, sorry about the brevity

Reply:
> Looks good other than one concern: [#4752 (comment)](https://github.com/paritytech/polkadot/pull/4752#discussion_r792459093).

Addressed it. We'll pick look back up to `max(last_finalized_block_number, new_leaf -MAX_BATCH_SCRAPE_ANCESTORS);`